src/main/resources/JavadocParser.g4

```diff
@@ -0,0 +1,104 @@
+parser grammar JavadocParser;
+
+options {
+    tokenVocab = JavadocLexer;
+}
+
+@parser::header {
+import java.util.Set;
+}
+
+
+@parser::members {
+    private static final Set<String> VOID_TAGS = Set.of(
+        "area", "base", "basefont", "br", "col", "frame", "hr",
+        "img", "input", "isindex", "link", "meta", "param"
+    );
+
+    public boolean isVoidTag() {
+        String tagName = _input.LT(2).getText();
+        return VOID_TAGS.contains(tagName.toLowerCase());
+    }
+}
+
+javadoc
+    : mainDescription (blockTag)* EOF;
+
+mainDescription: (NEWLINE | TEXT | inlineTag | htmlElement)*;
+
+inlineTag
+    : JAVADOC_INLINE_TAG_START
+     ( codeInlineTag | linkInlineTag | linkPlainInlineTag | customInlineTag)
+      JAVADOC_INLINE_TAG_END
+    ;
+
+codeInlineTag: CODE_LITERAL (TEXT|NEWLINE)*;
+
+linkPlainInlineTag: LINKPLAIN_LITERAL reference description?;
+
+linkInlineTag: LINK_LITERAL reference description?;
+
+reference
+        : qualifiedName (HASH memberReference)?
+        | (HASH memberReference)
+        ;
+
+qualifiedName: IDENTIFIER (DOT IDENTIFIER)*;
+
+memberReference: IDENTIFIER (LPAREN parameterList? RPAREN)?;
+
+parameterList: IDENTIFIER (COMMA IDENTIFIER)*;
+
+customInlineTag: CUSTOM_NAME description?;
+
+blockTag
+    : authorTag
+    | deprecatedTag
+    | returnTag
+    | parameterTag
+    | customBlockTag
+    ;
+
+authorTag: AUTHOR_LITERAL description;
+
+deprecatedTag: DEPRECATED_LITERAL description;
+
+returnTag: RETURN_LITERAL description;
+
+parameterTag: PARAM_LITERAL parameterName description;
```

Comment:
@nrmancuso So we are continuing the behavior that we will throw an exception on badly formed javadocs?

Example here would be `PARAM_LITERAL` with no `parameterName` or `description`.

Reply:
> recover from such issues

Can you expand on what this means? From our discussion, I was assuming we will keep the exception.

Reply:
ANTLR can fill in the gaps for us; if we want to be “not strict” in our parsing, it is better to recover than to throw exceptions.